testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/cdi/basic/DecoratorsTest.java

```diff
@@ -71,6 +74,19 @@
         return war;
     }
 
+    private ResteasyProviderFactory factory;
+    @Before
+    public void setup() {
+        // Create an instance and set it as the singleton to use
+        factory = ResteasyProviderFactory.newInstance();
```

Comment:
I believe _RegisterBuiltin.register(factory);_ should also be executed, in order to have a provider factory that behaves the expected way. WDYT? (this applies to all similar code blocks in this PR)